modules/router/src/Router.ts

```diff
@@ -461,4 +470,161 @@ export default class ConduitDefaultRouter extends ManagedModule<Config> {
     }
     ConduitGrpcSdk.Logger.log('Recovered routes');
   }
+
+  async patchMiddleware(
+    call: GrpcRequest<PatchAppMiddlewareRequest>,
+    callback: GrpcCallback<null>,
+  ) {
+    const { path, action, middleware } = call.request;
+    const moduleUrl = await this.grpcSdk.config.getModuleUrlByName(
+      call.metadata!.get('module-name')![0] as string,
+    );
+    if (!moduleUrl) {
+      return callback({
+        code: status.INTERNAL,
+        message: 'Something went wrong',
+      });
+    }
+    await this.internalPatchMiddleware(path, action, middleware, moduleUrl.url).catch(
+      e => {
+        return callback({
+          code: status.INTERNAL,
+          message: (e as Error).message,
+        });
+      },
+    );
+  }
+
+  async internalPatchMiddleware(
+    path: string,
+    action: string,
+    middleware: string[],
+    moduleUrl?: string,
+  ) {
+    let injected: string[] = [];
+    let removed: string[] = [];
+    if (!moduleUrl) {
+      moduleUrl = await this.grpcSdk.config.getModuleUrlByName('router').then(r => r.url);
```

Comment:
Add a comment clarifying this is about administrators modifying client API middlewares.